### DISCUSSION TITLE
API Token generated from Settings is not in JWT format

### DISCUSSION THREAD

**Discussion by abhisheksachdeva** [outline/outline#3419] [General]
Hi team, I am not sure if this is a bug, so posting it here.

I generated an `API_KEY` from [/settings/tokens](https://app.getoutline.com/settings/tokens) as mentioned [here](https://www.getoutline.com/developers#section/Authentication). It is in the format `token-name-some-hash`.

When I use the token in the header (`authorization`), I get this error
```
{"ok":false,"error":"authentication_required","status":401,"message":"Unable to decode JWT token"}
```

I checked browser cookies, where the `accessToken` is in the JWT format. The API works fine while using the `accessToken` from the browser.

Curl request for testing
```
curl https://app.getoutline.com/api/documents.list -X POST -H 'authorization: Bearer API_KEY' \
-H 'content-type: application/json' \
-H 'accept: application/json'
```

Am I missing anything? Shouldn't the generated token be in JWT format?

**Comment by tommoor** [comment]
The token name is not part of the API key, just use the hash – the UI could be clearer here I guess!

**Reply by abhisheksachdeva** [reply]
Thanks, that worked! 

Yes, a minor change in UI is required.

**Reply by almereyda** [reply]
In a first attempt, this could be as easy as writing the token below its description. Also if the token were written in a non-editable input box, the drag-select and double-click-select would probably not extend beyond its borders.

And just as examples from other applications:

GitHub does it in a similar way for the Client ID of an OAuth application.

![Bildschirmfoto von 2022-04-19 22-40-06](https://user-images.githubusercontent.com/1645308/164091927-57792ac1-0794-4734-bd79-cc907888e7af.png)

Yet since this is a secret, it should possibly only been shown once, and have a similar delete-and-recreate rule as the above example.

GitLab solves this in another way:

![Bildschirmfoto von 2022-04-19 22-41-47](https://user-images.githubusercontent.com/1645308/164092114-17fd0ce3-313d-4935-8cbb-9769e81604ed.png)

**Reply by abhisheksachdeva** [reply]
Sounds good.

Using a hard boundary (or `:`) between name and token would have helped. In my case, the name had a hyphen that caused the confusion.
![image](https://user-images.githubusercontent.com/9962817/164157469-3db3cb36-e976-453b-9efe-ae4d38e4ab90.png)
